test/noShiftGradientTest.html

```diff
@@ -0,0 +1,134 @@
+      or testBuild.bat-->
+  <script src="../build/spot.js"></script>
+
+  <!--Mocha setup stuff (1/2)-->
+  <meta charset="UTF-8">
+  <script src="../lib/test/mocha.min.js"></script>
+  <script src="../lib/test/chai.min.js"></script>
+  <link rel="stylesheet" type="text/css" href="../lib/test/mocha.css">
+  <!--mocha setup stuff to be continued in document body-->
+  
+</head>
+
+<body style="padding-left: 5%; padding-right: 5%; padding-top: 20px">
+  <h2>SPOT Automated Testing</h2>
+  <script>
+    mocha.setup("bdd"); //brings "describe", "it", etc. into global namespace
+        //mocha.checkLeaks();
+        var assert = chai.assert; //no one wants to type out "chai.assert" every time
+</script>
+
+<div id="mocha">
+    No Shift Gradient Automated Test
+</div>
+  <script>
+    // example with moving x
+    console.log("a b c d X")
+    var sorder1 = ["a", "b", "c", "d", "x"];
+    var porder1 = ["a", "b", "c", "d", "x"];
+    console.log("a b c d X: ", noShiftGradient(sorder1, porder1), " expected: 0")
+    porder2 = ["a", "b", "c", "x", "d"];
+    console.log("a b c X d: ", noShiftGradient(sorder1, porder2), " expected: 1")
+    porder3 = ["a", "b", "x", "c", "d"];
+    console.log("a b X c d: ", noShiftGradient(sorder1, porder3), " expected: 2")
+    porder4 = ["a", "x", "b", "c", "d"];
+    console.log("a X b c d: ", noShiftGradient(sorder1, porder4), " expected: 3")
+    porder5 = ["x", "a", "b", "c", "d"];
+    console.log("X a b c d: ", noShiftGradient(sorder1, porder5), " expected: 4")
+
+    // example with potential movers x,y
+    console.log("a b c d X Y")
+    sorder2 = ["a", "b", "c", "d", "x", "y"];
+    porder6 = ["x", "a", "b", "c", "d", "y"];
+    console.log("X a b c d Y: ", noShiftGradient(sorder2, porder6), " expected: 4")
+    porder7 = ["y", "a", "b", "c", "d", "x"];
+    console.log("Y a b c d X: ", noShiftGradient(sorder2, porder7), " expected: 5")
+    porder8 = ["x", "y", "a", "b", "c", "d"];
+    console.log("X Y a b c d: ", noShiftGradient(sorder2, porder8), " expected: 8")
+    porder9 = ["y", "x", "a", "b", "c", "d"];
+    console.log("Y X a b c d: ", noShiftGradient(sorder2, porder9), " expected: 9")
+
+    // additional test cases
+    console.log("a b c d X")
+    sorder3 = ["a", "b", "c", "d", "x"];
+    pordera = ["x", "d", "c", "b", "a"];
+    console.log("X d c b a: ", noShiftGradient(sorder3, pordera), " expected: 10")
+    porderb = ["b", "a", "c", "d", "x"];
+    console.log("b a c d x: ", noShiftGradient(sorder3, porderb), " expected: 1")
+    porderc = ["c", "a", "b", "d", "x"];
+    console.log("c a b d x: ", noShiftGradient(sorder3, porderc), " expected: 2")
+    porderd = ["d", "a", "b", "c", "x"];
+    console.log("d a b c X: ", noShiftGradient(sorder3, porderd), " expected: 3")
+
+    //automated test
+    describe('Moving X,  "a b c d X"', function() {
+            it('"a b c d X:", noShiftGradient(sorder, porder)', function() {
```

Comment:
Would it work to declare all these variables closer to where they are being used? I think that would improve the interpretability of this file.

Reply:
Declare / define porder inside the it() block so that it's easy to see what its value is in the browser.